From 24bcf7068b6ebf58fe84365c6a91e1e8b5dd861d Mon Sep 17 00:00:00 2001
From: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Date: Wed, 13 Dec 2017 13:11:42 +0530
Subject: [PATCH 1443/2328] Soundwire: Check for pointer validity

In sdw_register_slave_capabilities(), memory allocation is not
checked for null at few places. Check for pointer validity and
return -ENOMEM if allocation fails.

Change-Id: Ic28884385c8c210b24f306a0b436252e484f3e24
Review-by: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 drivers/sdw/sdw.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/sdw/sdw.c b/drivers/sdw/sdw.c
index 4be1afc..ee09aed 100644
--- a/drivers/sdw/sdw.c
+++ b/drivers/sdw/sdw.c
@@ -2087,6 +2087,9 @@ int sdw_register_slave_capabilities(struct sdw_slave *sdw,
 	slv_cap->sdw_dpn_cap = devm_kzalloc(&sdw->dev,
 			((sizeof(struct sdw_slv_dpn_capabilities)) *
 			cap->num_of_sdw_ports), GFP_KERNEL);
+	if (!slv_cap->sdw_dpn_cap)
+		return -ENOMEM;
+
 	for (i = 0; i < cap->num_of_sdw_ports; i++) {
 		dpn_cap = &cap->sdw_dpn_cap[i];
 		slv_dpn_cap = &slv_cap->sdw_dpn_cap[i];
@@ -2142,6 +2145,9 @@ int sdw_register_slave_capabilities(struct sdw_slave *sdw,
 		slv_dpn_cap->mode_properties = devm_kzalloc(&sdw->dev,
 				((sizeof(struct port_audio_mode_properties)) *
 				dpn_cap->num_audio_modes), GFP_KERNEL);
+		if (!slv_dpn_cap->mode_properties)
+			return -ENOMEM;
+
 		for (j = 0; j < dpn_cap->num_audio_modes; j++) {
 			prop = &dpn_cap->mode_properties[j];
 			slv_prop = &slv_dpn_cap->mode_properties[j];
-- 
2.7.4

